heron/api/src/scala/com/twitter/heron/streamlet/scala/impl/BuilderImpl.scala

```diff
@@ -0,0 +1,39 @@
+//  Copyright 2018 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+package com.twitter.heron.streamlet.scala.impl
+
+import com.twitter.heron.streamlet.scala.{Builder, Source, Streamlet}
+import com.twitter.heron.streamlet.scala.converter.ScalaToJavaConverter
+
+
+class BuilderImpl extends Builder {
+
+  override def newSource[R](supplierFn: () => R): Streamlet[R] = {
```

Comment:
BuilderImpl level, we need to call Java Builder `` functions. Currently, Java BuilderImpl creates a `List<StreamletImpl<?>> sources` and keeps `Sources, Transformations and Sink` to build `topology` so this is why important and for example, this function needs to be as follows:
```
override def newSource[R](supplier: () => R): Streamlet[R] = {
    val serializableSupplier = toSerializableSupplier[R](supplier)
    val newJavaStreamlet = builder.newSource(serializableSupplier) //Java Builder
    StreamletImpl.toScalaStreamlet[R](newJavaStreamlet)
  }
```

Reply:
Changed for both methods to leverage java builder

Reply:
Changed